rdkPlugins/Gamepad/source/GamepadPlugin.cpp

```diff
@@ -0,0 +1,176 @@
+/*
+* If not stated otherwise in this file or this component's LICENSE file the
+* following copyright and licenses apply:
+*
+* Copyright 2020 Sky UK
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "GamepadPlugin.h"
+
+// #include <linux/limits.h>
+// #include <sys/mount.h>
+// #include <sys/stat.h>
+// #include <sys/types.h>
+// #include <fcntl.h>
+// #include <unistd.h>
+// #include <mntent.h>
+#include <iostream>
+#include <sstream>
+
+REGISTER_RDK_PLUGIN(GamepadPlugin);
+
+GamepadPlugin::GamepadPlugin(std::shared_ptr<rt_dobby_schema> &containerConfig,
+                     const std::shared_ptr<DobbyRdkPluginUtils> &utils,
+                     const std::string &rootfsPath)
+    : mName("Gamepad"),
+      mContainerConfig(containerConfig),
+      mUtils(utils)
+{
+    AI_LOG_FN_ENTRY();
+    AI_LOG_FN_EXIT();
+}
+
+unsigned GamepadPlugin::hookHints() const
+{
+    return (IDobbyRdkPlugin::HintFlags::PostInstallationFlag);
+}
+
+// Begin Hook Methods
+
+/**
+ * @brief Dobby Hook - run in host namespace *once* when container bundle is downloaded
+ */
+bool GamepadPlugin::postInstallation()
+{
+    AI_LOG_FN_ENTRY();
+
+    // 1. add devices
+    rt_config_linux_resources* schemaResources = mContainerConfig->linux->resources;
```

Comment:
Please create add_controller_device (or similar) function and use it to separate the logic

Reply:
Done